From 3b30508480af77b589ba906b4592e7423ff07178 Mon Sep 17 00:00:00 2001
From: Zhao Liang <leo.zhao@intel.com>
Date: Mon, 23 Sep 2013 10:29:03 +0800
Subject: [PATCH] [PORT FROM MAIN] Enable video cross process buffer sharing

Setup service "media.IntelBufferSharing" for sharing peers

Change-Id: I217dc10cd7a37e0aacf93a887eb0397c3a1abbe3
Category: feature differentiation
Domain: Video.Media-Framework
Origin: internal
Signed-off-by: Zhao Liang <leo.zhao@intel.com>
Reviewed-on: http://android.intel.com:8080/133545
Reviewed-by: Yuan, Shengquan <shengquan.yuan@intel.com>
Tested-by: Post, DavidX J <davidx.j.post@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>

diff --git a/media/mediaserver/Android.mk b/media/mediaserver/Android.mk
index 4f72759..25cdaee 100644
--- a/media/mediaserver/Android.mk
+++ b/media/mediaserver/Android.mk
@@ -25,6 +25,11 @@ LOCAL_SHARED_LIBRARIES := \
 	liblog \
 	libbinder
 
+ifeq ($(INTEL_VIDEO_XPROC_SHARING),true)
+LOCAL_SHARED_LIBRARIES += libintelmetadatabuffer
+LOCAL_C_INCLUDES += $(TARGET_OUT_HEADERS)/libmix_videoencoder
+endif
+
 LOCAL_STATIC_LIBRARIES := \
 	libregistermsext
 
diff --git a/media/mediaserver/main_mediaserver.cpp b/media/mediaserver/main_mediaserver.cpp
index 4659996..4c145b5 100644
--- a/media/mediaserver/main_mediaserver.cpp
+++ b/media/mediaserver/main_mediaserver.cpp
@@ -37,6 +37,9 @@
 #ifdef AUDIO_LISTEN_ENABLED
 #include "ListenService.h"
 #endif
+#ifdef INTEL_VIDEO_XPROC_SHARING
+#include "IntelMetadataBuffer.h"
+#endif
 
 using namespace android;
 
@@ -144,6 +147,9 @@ int main(int argc, char** argv)
         ListenService::instantiate();
 #endif
         AudioPolicyService::instantiate();
+#ifdef INTEL_VIDEO_XPROC_SHARING
+        IntelBufferSharingService::instantiate();
+#endif
         registerExtensions();
         ProcessState::self()->startThreadPool();
         IPCThreadState::self()->joinThreadPool();
